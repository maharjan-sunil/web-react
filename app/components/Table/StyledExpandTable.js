import styled from 'styled-components';
import Table from '@bootstrap-styled/v4/lib/Table';

const StyledExpandTable = styled(Table)`
  background: #fff !important;
  border: 1px solid #dee2e6;
  & tbody:hover tr.main-row {
    background: rgb(27, 81, 146, 0.15);
  }
  & tbody tr.active {
    background: rgb(27, 81, 146, 0.15);
  }
  &.table {
    .tr_icon {
      width: 30px;
    }
    .tr_body {
      display: none;
    }
    tbody.expand {
      .tr_body {
        display: table-row;
      }
      .sub-row{
        &:hover{
          background: rgb(100, 81, 190, 0.25);
        }
      }
    }
`;

export default StyledExpandTable;
