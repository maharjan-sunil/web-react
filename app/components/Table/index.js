import styled from 'styled-components';
import Table from '@bootstrap-styled/v4/lib/Table';

const StyledTable = styled(Table)`
  background: #fff !important;
  border: 1px solid #dee2e6;
  & tbody tr:hover {
    background: rgb(27, 81, 146, 0.15);
  }
  & tbody tr.active {
    background: rgb(27, 81, 146, 0.15);
  }
`;

export default StyledTable;
