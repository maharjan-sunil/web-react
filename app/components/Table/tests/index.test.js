import React from 'react';
import { render } from 'react-testing-library';

import Table from '../index';

describe('<Table />', () => {
  it('should render an <Table> tag', () => {
    const { container } = render(<Table />);
    const element = container.firstElementChild;
    expect(element.tagName).toEqual('TABLE');
  });

  it('should match snapshot ', () => {
    const { container } = render(<Table />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
