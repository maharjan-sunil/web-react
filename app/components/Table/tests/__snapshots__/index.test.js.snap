// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> should match snapshot  1`] = `
<table
  class="Table-oyilxr-0 Table__StyledTable-sc-242oap-0 lldlQi table"
/>
`;
